doc/reference/eventengine.rst

```diff
@@ -38,6 +38,117 @@ of alerts. The default configuration looks somewhat like this:
+configuration file :file:`severity.yml`. Any rules present in this file will be
+processed to set or modify the existing severity of any matching alert that is
+generated.
+
+Configuring ``severity.yml``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is an example severity configuration:
+
+.. code-block:: yaml
+   :caption: severity.yml
+
+   ---
+   default-severity: 3
+   rules:
+       - alert_type: boxDown
+         severity: 2
+         rules:
+             - netbox.category.id: GSW
+               severity: 1
+             - netbox.category.id: GW
+               severity: 1
+
+       - alert_type: boxDownWarning
+         severity: 5
+
+       - netbox.organization.id: foobar
+         severity: '+2'
+
+This configuration starts off by assigning a *default severity* level of **3**
+to every alert that :program:`Event Engine` generates, regardless of what the
+original severity value of the event was.
+
+Then follows a list of rules that will be processed *in the order they appear*
+in the file. Each rule consists of:
+
+- One or more alert attribute match expressions.
+- One severity value modification expression to be applied to an alert that
+  matches the attribute expressions.
+- Optionally, a sub-list of more rules to further apply to any alert that
+  matched the expressions of this rule.
+
+The first example rule will match any alert whose ``alert_type`` value equals
+``boxDown`` (NAV's alert type for a lasting "box is unreachable" incident). Any
+such alert will be assigned a severity level of **2**.  Furthermore, the rule
+lists two additional sub-rules to ensure that if the ``boxDown`` alert was
+issued for any netbox (IP Device) whose category is a router (a category id of
+either ``GSW`` or ``GW``), the severity is set to the most critical level of
+**1**.
+
+The second top-level example rule will match any alert whose type is
+``boxDownWarning``, and set its severity to the least critical level of
+**5**. This is the stateless early warning the :program:`Event Engine` issues a
+few minutes before declaring a stateful ``boxDown``. It is safe to consider
+this type of alert as only *informational*.
+
+The final top-level example rule will match any alert whose associated netbox
+(IP Device) is owned by the organizational id ``foobar``. This rule uses a
+*severity modifier expression* of ``+2``, which will add ``2`` to the current
+alert's existing severity value.
+
+In summary, if a ``boxDown`` alert is dispatched for a router in your network,
+this rule set will ensure its severity is set to **1**. However, if the router
+belongs to your less important ``foobar`` department, two severity levels will
+be deducted, and the alert comes out with a severity of **3**.
+
+
+Modifier expressions
+~~~~~~~~~~~~~~~~~~~~
+
+There are three types of supported severity modifier expressions for use in rules:
+
+1. Absolute values: An integer in the range 1-5 will *replace* a matching
```

Comment:
What happens with integers outside this range?
Will they be aligned to the closest possible valid value, or will an error occur, or something else?

Reply:
The comment about arithmetic outside the 1-5 range is also valid here. Attempting to assign a value `<1` will result in `1`, assigning something `>5` results in 5 - no errors are raised.

Reply:
There's a specific mathematical term for this (rings, maybe?) but no need to scare people.

Reply:
I simply reduced the statement to say that Event Engine silently makes sure no value assigned, absolute or relative, will exceed the valid range. Unless the reader is a mathematician, I don't think we need to be wordy to explain that a value of 42 would become 5 and a value of -27 becomes 1 ;-)